app/controllers/foreman_tasks/tasks_controller.rb

```diff
@@ -31,9 +31,12 @@ def sub_tasks
 
     def cancel_step
       task = find_dynflow_task
-      flash[:info] = _('Trying to cancel step %s') % params[:step_id]
-      ForemanTasks.dynflow.world.event(task.external_id, params[:step_id].to_i, ::Dynflow::Action::Cancellable::Cancel).wait
-      redirect_to foreman_tasks_task_path(task)
+      result = ForemanTasks.dynflow.world.event(task.external_id, params[:step_id].to_i, ::Dynflow::Action::Cancellable::Cancel).wait
```

Comment:
It could happen that delivery of the event takes too long and before `ForemanTasks.dynflow....wait` returns, the http request times out. Will the frontend be able to deal with it?

Reply:
no, one min =>failure ):
So we can just show 'trying' toast or also add a notification like in the bulk action, what do you think is better?

Reply:
I'd say it is fine the way it is now